src/openloco/windows/promptbrowsewnd.cpp

```diff
@@ -639,20 +640,27 @@ namespace openloco::ui::prompt_browse
             auto directory = fs::path((char*)_directory);
             if (fs::is_directory(directory))
             {
-                for (auto& f : fs::directory_iterator(directory))
+                try
                 {
-                    bool isDirectory = fs::is_directory(f);
-                    if (!isDirectory)
+                    for (auto& f : fs::directory_iterator(directory))
                     {
-                        auto extension = f.path().extension().u8string();
-                        if (!utility::iequals(extension, filterExtension))
+                        auto isDirectory = f.is_directory();
```

Comment:
Why was this changed from bool to auto? (Perhaps a leftover from earlier debugging?)

I suggest we change it back:
```suggestion
                        bool isDirectory = f.is_directory();
```

Reply:
I see you're not an almost always auto person. Sure I'll change it back. I was also thinking we should actually if else check it is a file as there is a bit in-between from files to folders.

Reply:
Yeah, I like the use of `auto` for template-heavy types, but dislike it for primitive types.

I wouldn't mind the checks be changed as you suggest. To what extent are symlinks transparent to `is_directory` and `is_file`, though? I would prefer we keep supporting them.

Reply:
I'm not too sure how well it currently works. I'd have to run some tests. Technically there is also sockets and other. Symlinks, hard links and junctions all could have different interactions that cause issues.

Reply:
> I see you're not an almost always auto person

I am an always auto person. Always using auto where you can is just a simpler rule than deciding on a case by case basis. That said I never argue style choices like this as I believe it should just be up to the developer of that bit of code.

Reply:
I've now rejigged it so that it adds it to the list only if it is a directory or a regular file. I checked to see what happens with symlinks. They get listed just like a regular directory from my testing.

Reply:
I'd like to test it on Linux, but I imagine it behaves similarly.